server.go

```diff
@@ -282,12 +282,15 @@ func (s *Server) sendResponse(t transport.ServerTransport, stream *transport.Str
 }
 
 func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.Stream, srv *service, md *MethodDesc) (err error) {
+	ctx, cancel := context.WithCancel(stream.Context())
```

Comment:
Interesting.  trasport.Stream has a cancel field to cancel its Context, but I cannot see when it's called.  Still hunting.

I need this change for my gRPC demo, but I can pull it out into a separate PR in parallel with this one.